src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,150 @@
+        user.setLogin("testuser1");
+        user.setPassword("password1231");
+        user.setAge(20);
+
+        User registeredUser = registrationService.register(user);
+
+        Assertions.assertNotNull(registeredUser);
+        Assertions.assertNotNull(registeredUser.getId());
+        Assertions.assertEquals(user.getLogin(), registeredUser.getLogin());
+        Assertions.assertEquals(user.getPassword(), registeredUser.getPassword());
+        Assertions.assertEquals(user.getAge(), registeredUser.getAge());
+    }
+
+    @Test
+    public void register_InvalidUserRegistration_ExceptionThrown() {
+        User user = new User();
+        user.setLogin("short");
+        user.setPassword("password123");
+        user.setAge(20);
+
+        InvalidUserException exception = Assertions.assertThrows(InvalidUserException.class, () -> {
+            registrationService.register(user);
+        });
+
+        Assertions.assertEquals(INVALID_DATA, exception.getMessage());
+    }
+
+    @Test
+    public void register_InvalidUserPassLen_ExceptionThrown() {
+        User user = new User();
+        user.setLogin("testuser");
+        user.setPassword("short");
+        user.setAge(20);
+
+        InvalidUserException exception = Assertions.assertThrows(InvalidUserException.class, () -> {
+            registrationService.register(user);
+        });
+
+        Assertions.assertEquals(INVALID_DATA, exception.getMessage());
+    }
+
+    @Test
+    public void register_InvalidUserAge_ExceptionThrown() {
+        User user = new User();
+        user.setLogin("testuser");
+        user.setPassword("password123");
+        user.setAge(17);
+
+        InvalidUserException exception = Assertions.assertThrows(InvalidUserException.class, () -> {
+            registrationService.register(user);
+        });
+
+        Assertions.assertEquals(INVALID_DATA, exception.getMessage());
+    }
+
+    @Test
+    public void register_DuplicateUser_ExceptionThrown() {
+        User user = new User();
+        user.setLogin("testuser");
+        user.setPassword("password123");
+        user.setAge(20);
+
+        StorageDao storageDao = new StorageDaoImpl();
+        storageDao.add(user);
+
+        InvalidUserException exception = Assertions.assertThrows(InvalidUserException.class, () -> {
+            registrationService.register(user);
+        });
+
+        Assertions.assertEquals(EXISTS_MSG, exception.getMessage());
+    }
+
+    @Test
+    public void register_NullUser_ExceptionThrown() {
+        User user = null;
+
+        InvalidUserException exception = Assertions.assertThrows(InvalidUserException.class, () -> {
+            registrationService.register(user);
+        });
+
+        Assertions.assertEquals(INVALID_DATA, exception.getMessage());
+    }
+
+    @Test
+    public void register_NullLogin_ExceptionThrown() {
+        User user = new User();
+        user.setLogin(null);
+        user.setPassword("password123");
+        user.setAge(20);
```

Comment:
use constants
move user to test class field and set it's fields in the BeforeEach method to avoid code duplicating